src/server.js

```diff
@@ -498,6 +505,7 @@ function performAuthentication(data) {
 		socket.emit("configuration", getClientConfiguration());
 
 		client.ip = getClientIp(socket);
+		client.language = getClientLanguage(socket);
```

Comment:
This helper function isn't really needed, you can put `socket.handshake.headers["accept-language"]` directly here.

However I would either perform some kind of validation on the string, or at least limit its length to 50 or something.

Reply:
> However I would either perform some kind of validation on the string

I was considering this but decided not to because the header is set entirely by the browser (I'd like to think that browsers sets the header correctly)

Not sure this is how one wants to determine user's language, hence the helper function.

Reply:
I primarily want to prevent abuse here, so at least limit the strings length. And perhaps only allow ASCII characters in the string.

Reply:
You mean something like this:

```diff
diff --git a/src/server.js b/src/server.js
index 74ac4964..ef78aab7 100644
--- a/src/server.js
+++ b/src/server.js
@@ -181,8 +181,9 @@ module.exports = function() {
 };

 function getClientLanguage(socket) {
-       if ("accept-language" in socket.handshake.headers) {
-               return socket.handshake.headers["accept-language"];
+       const acceptLanguage = socket.handshake.headers["accept-language"]
+       if (typeof acceptLanguage === 'string' && /^[\x00-\x7F]{1,50}$/.test(acceptLanguage)) {
+               return acceptLanguage
        }
        return null;
 }

```

Reply:
That'll most likely do, yeah.

Reply:
@xPaw, helper function could help with unit testing, if we wanted to do that :man_shrugging: